### ISSUE TITLE
IntelliJ plugin doesn't respect max line length setting.

### ISSUE THREAD

**Issue by pardom** [facebook/ktfmt#351]
The IntelliJ plugin does not provide an option to consume `max_line_length` as `FormattingOptions.maxWidth` in `UiFormatter`. Thus we cannot synchronize the IDE with the overridden setting in our build script:

```
spotless {
    kotlin {
        ktfmt("0.40").kotlinlangStyle().configure { optionsConfiguration ->
            optionsConfiguration.setMaxWidth(120)
        }
    }
}
```

`KtfmtConfigurable.form` should provide an option to respect the `Hard wrap at` setting in `Preferences > Editor > Code Style` or `max_line_length` option in `.editorconfig`.

**Comment by twomoose** [comment]
Definitely interested to see if this is something that would be considered. The discrepancy between IDE and linter is one of the main issues my team has with ktlint (our current style tool). We're considering ktfmt to replace it but I can't convince others to change our max line width to 100 (as required by ktfmt). Spotless gets around this but then we have the issue of the IDE plugin clashing with build script tooling.

**Comment by runenielsen** [comment]
This would be very useful.

**Comment by hick209** [comment]
@mateherber-jt, oh so the issue persist even at 0.50? Interesting...
Long shot from my end, as I don't really know much about how Spotless and IJ formatting works, but have you tried using [`.editorconfig`](https://github.com/editorconfig/editorconfig/wiki/EditorConfig-Properties#max_line_length)? I wonder if that's the missing piece of the puzzle

**Comment by greyhairredbear** [comment]
As far as I understand from working on the plugin and looking into the ktfmt step for spotless, both use ktfmt as a direct dependency, but configure it in two different locations (IJ settings vs gradle file). 

I think the easiest way to implement this would be allowing to set that value directly in the config form - if that is something that should be configurable. Not sure how much effort it would require to implement reading the max line length value from an `.editorconfig` file or from the IJ code style settings. Also, I'm unsure if that would actually be sensible. The less places you have to go to configure ktfmt, the better.

**Comment by leourbina7r** [comment]
I'm running into this same issue. It seems like the Intellij plugin uses 100 for max_line_limit regardless of the `.editorconfig`.

**Comment by rock3r** [comment]
Fixed in #503

**Comment by hick209** [comment]
Should be fixed by https://github.com/facebook/ktfmt/pull/503
